HIP/hip-1127.md

```diff
@@ -0,0 +1,93 @@
+---
+hip: 1127
+title: Transaction Cleanup
+author: Michael Heinrichs <@netopyr>
+working-group: Richard Bair <@rbair>, Jasper Potts <@jasperpotts>
+requested-by: Richard Bair <@rbair>
+type: "Standards Track"
+category: "Service"
+needs-council-approval: "Yes"
+status: "Draft"
+created: 2025-02-17
+updated: 2025-02-17
+requires: 
+replaces: 
+superseded-by: 
+---
+
+## Abstract
+There are three ways to store the `TransactionBody` and the `SignatureMap` in a `Transaction` which is unnecessarily complex.
+This HIP defines a single standard of how these fields should be stored and how to handle transactions that do not follow the new normalized format.
+
+## Motivation
+Every program that parses transactions and extracts `TransactionBody` and `SignatureMap` requires special code to deal with all possible ways to store these fields. In addition, all but one possibility defines part of the data as byte arrays that need to be deserialized in an extra step.
+This causes unnecessary complexity that we want to avoid.
+
+## Rationale
+To simplify the handling of transactions, we selected one of the possible ways to store the transaction data.
+
+All incoming transactions that do not follow the standard format will be adjusted accordingly.
+The required modifications will be done so that they do not break the signatures.
+
+We chose `Transaction.body` and `Transaction.sigMap` as the standard format. 
+This simplifies transaction parsing because `Transaction.body` is specified as a `TransactionBody` object, which allows us to parse a transaction in one go.
+
+## User stories
+1. As a transaction sender, I want to be able to send transactions using all possible formats so that I am not forced to rewrite the calling code (which may be impossible).
+2. As a record stream/block stream consumer, I want a simple way to parse transactions and extract data.
+3. As a transaction signer, I want to be sure that my signatures can always be validated, even if transactions are modified to follow the normalized format.
+
+## Specification
+The protobuf definition of the `Transaction` object will not change directly. 
+However, we "un-deprecate" `body` and `sigMap`, while marking `signedTransactionBytes` as deprecated.
+
+```protobuf
+message Transaction {
+
+    TransactionBody body = 1;
```

Comment:
How will this work for AtomicBatch? You'd have to somehow use the root transaction bytes and somehow split it into the bytes of each inner? How will you detect the start and end of inners especially in the presence of unknown fields?

Reply:
Looks like it's pretty [straightforward](https://github.com/hiero-ledger/hiero-consensus-node/blob/main/hedera-node/hedera-app/src/main/java/com/hedera/node/app/util/ProtobufUtils.java#L43).